news/templatetags/news_tags.py

```diff
@@ -12,15 +11,9 @@ def color(val):
 
 @register.filter(name="past")
 def past(timeplace_set):
-    return timeplace_set.filter(Q(end_date=timezone.now().date(), end_time__lt=timezone.now().time()) |
-                                Q(end_date__lt=timezone.now().date()) |
-                                Q(end_date=None, end_time__lt=timezone.now().time(), start_date=timezone.now().date()) |
-                                Q(end_date=None, start_date__lt=timezone.now().date()))
+    return timeplace_set.filter(end_time__lt=timezone.now())
 
 
 @register.filter(name="future")
 def future(timeplace_set):
-    return timeplace_set.filter(Q(end_date=timezone.now().date(), end_time__gt=timezone.now().time()) |
-                                Q(end_date__gt=timezone.now().date()) |
-                                Q(end_date=None, end_time__gt=timezone.now().time(), start_date=timezone.now().date()) |
-                                Q(end_date=None, start_date__gt=timezone.now().date()))
+    return timeplace_set.filter(end_time__gt=timezone.now())
```

Comment:
One of `past()` and `future()` should probably be `lte` or `gte`, like you changed with `TimePlaceManager.past()` and `TimePlaceManager.future()` 🙂